### PR TITLE
Enable nullable on Kestrel generated code

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/issues/5680

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs [hunk 1]
```diff
@@ -12816,7 +12818,7 @@ internal unsafe void CopyToFast(ref BufferWriter<PipeWriter> output)
                         {
                             tempBits ^= 0x8000000000000000L;
                             output.Write(HeaderBytes.Slice(640, 18));
-                            output.WriteNumeric((ulong)ContentLength.Value);
+                            output.WriteNumeric((ulong)ContentLength.GetValueOrDefault());
```
Comment (JamesNK) [inline src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs:12821]: FYI GetValueOrDefault method is slightly faster. It doesn't check whether a value is set, unlike Value property.

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs [hunk 2]
```diff
@@ -369,137 +371,137 @@ private void ExtraFeatureSet(Type key, object value)
...
+                feature = (TFeature?)(ExtraFeatureGet(typeof(TFeature)));
             }
 
             if (feature == null)
             {
                 feature = ConnectionFeatures.Get<TFeature>();
             }
 
-            return feature;
+            return feature!;
```
Comment (JamesNK) [inline src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs:504]: Should `IFeatureCollection.Get<T>()` return a nullable value? It's technically correct, but could harm usability of the API.

@pranavkm is there a reason it was left as non-null in https://github.com/dotnet/aspnetcore/pull/22337?

Comment (pranavkm) [reply src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs:504]: I think it's intended to return null if the feature does not exist (`ISessionFeature`, `IEndpointFeature` etc are optional). What would be useful is to introduce a `GetRequiredFeature` for cases like this where we expect the feature to exist.

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs [hunk 3]
```diff
@@ -369,137 +371,137 @@ private void ExtraFeatureSet(Type key, object value)
                 }
                 else
                 {
-                    ExtraFeatureSet(key, value);
+                    ExtraFeatureSet(key, value!); // TODO: What happens if you set an extra feature with a null value?
```
Comment (pranavkm) [inline src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs:374]: Do you plan to file an issue to figure this out?

**Review by pranavkm — APPROVED** [review]

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs (hunk 3 above)
Comment (JamesNK) [reply src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs:391]: I'm going to fix it in this PR

File: src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs [hunk 4]
```diff
@@ -104,33 +106,50 @@ private object ExtraFeatureGet(Type key)
             return null;
         }
 
-        private void ExtraFeatureSet(Type key, object value)
+        private void ExtraFeatureSet(Type key, object? value)
```
Comment (JamesNK) [inline src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs:109]: Previously calling ExtraFeatureSet with a null value would leave a null entry in MaybeExtra. These entries would still be returned when foreaching over feature collection.

Now null extra features are completely removed.

**Pull request merged**